### PR TITLE
migrate getKey to getSchemaKeyFromHeading with nicer API surface

### PR DESCRIPTION
**Pull request by camjc**
## Summary

Make a new util that matches headings to schema keys with more control (guarantee single match, slugify before matching) and also presents a nicer API surface.

### CODE REVIEW

File: src/shared/sources/in/index.js [hunk 1]
```diff
@@ -5,13 +5,13 @@ const transform = require('../_lib/transform.js')
...
 
-const labelFragmentsByKey = [
-  { state: 'name of state' },
-  { deaths: 'death' },
-  { cases: 'total confirmed cases' },
-  { recovered: 'cured' },
-  { discard: 's. no.' }
-]
+const schemaKeysByHeadingFragment = {
+  'name of state': 'state',
```
Comment (camjc) [inline src/shared/sources/in/index.js:9]: left side is a fragment of whats in the heading (will slugify and check the slug of this is uniquely inside the slug of the heading), right side is something in our schema that it maps to (or null when we want to throw out a column).

File: src/events/scraper/run-scraper/scraper-helpers/get-schema-key-from-heading.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  for (const key in object) {
+    if (predicate(object[key], key)) {
+      obj[key] = object[key]
+    }
+  }
+  return obj
+}
+
+const assertAllValuesAreInSchema = (schemaKeysByHeadingFragment) => {
+  Object.values(schemaKeysByHeadingFragment).forEach(potentialKey => assert(schemaKeys.some(validKey => validKey === potentialKey), `Invalid value in schemaKeysByHeadingFragment: ${potentialKey}`))
```
Comment (ryanblock) [inline src/events/scraper/run-scraper/scraper-helpers/get-schema-key-from-heading.js:32]: This line is 195 characters. Please don't make me add linting rules. :)

Can we pls break into multiple lines with clear actions?

File: src/events/scraper/run-scraper/scraper-helpers/get-schema-key-from-heading.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+}
+
+const getSchemaKeyFromHeading = ({ heading, schemaKeysByHeadingFragment }) => {
+  assertAllValuesAreInSchema(schemaKeysByHeadingFragment)
+  const slugHeading = slugify(heading, slugifyOptions)
+  const foundItems = pickBy(schemaKeysByHeadingFragment, (schemaKey, headingFragment) => slugHeading.includes(slugify(headingFragment, slugifyOptions)))
```
Comment (ryanblock) [inline src/events/scraper/run-scraper/scraper-helpers/get-schema-key-from-heading.js:38]: Another that would be much more readable in multiple lines, please!

File: src/events/scraper/run-scraper/scraper-helpers/get-schema-key-from-heading.js (hunk 2 above)
Comment (camjc) [reply src/events/scraper/run-scraper/scraper-helpers/get-schema-key-from-heading.js:32]: Can we please add linting rules? I'm so used to prettier at 120char that I don't normally have to think about line lengths 😬

Comment (ryanblock) [reply src/events/scraper/run-scraper/scraper-helpers/get-schema-key-from-heading.js:32]: I was only kidding, but feel free to add! A hard 120 char would still get pretty far, but I still find it difficult to read, compared to something more verbose but comprehensible (to me) like:
```
const foundItems = pickBy(schemaKeysByHeadingFragment, (schemaKey, headingFragment) => {
  const heading = slugify(headingFragment, slugifyOptions)
  return slugHeading.includes(heading)
})
```

**Pull request merged**